### PR TITLE
Implement fast tls for s390x

### PR DESCRIPTION
**Pull request by nealef**
Reimplements fast tls for s390x using the new structure.

### CODE REVIEW

File: mono/mini/mini-s390x.h [hunk 1]
```diff
@@ -62,6 +62,7 @@ typedef struct
 #define MONO_ARCH_HAVE_INVALIDATE_METHOD		1
 #define MONO_ARCH_HAVE_OP_GENERIC_CLASS_INIT		1
 #define MONO_ARCH_HAVE_SETUP_ASYNC_CALLBACK		1
+#define MONO_ARCH_HAVE_FAST_TLS				1
```
Comment (BrzVlad) [inline mono/mini/mini-s390x.h]: We no longer use this define

File: mono/utils/mono-tls.c [hunk 2]
```diff
@@ -116,6 +116,44 @@
 
 #define MONO_THREAD_VAR_OFFSET(var,offset) __asm ("     ldr     %0, 1f; b 2f; 1: .word " #var "(tpoff); 2:" : "=r" (offset))
 
+#elif defined(__s390x__)
```
Comment (BrzVlad) [inline mono/utils/mono-tls.c]: Is TARGET_S390X well defined on this target ? Could we use it instead ?

**Comment by nealef** [comment]
Suggested fixes pushed.

**Comment by nealef** [comment]
Errors in checks look unrelated - especially since these are architecture specific changes. I have pushed changes for the fixes you recommended.

**Comment by BrzVlad** [comment]
@nealef could you please squash the 2 commits and then we can merge

**Comment by nealef** [comment]
Will do

-------- Original message --------

**Comment by BrzVlad** [comment]
@nealef apparently github has a squash button, merged

**Comment by nealef** [comment]
Handy to know! Thanks

**Pull request merged**